src/qml/main.qml

```diff
@@ -24,16 +24,40 @@ import harbour.kaidan 1.0
 
 Kirigami.ApplicationWindow {
 	id: root
-	visible: true
-
-	// first page: login, the roster page will be opened after it
-	pageStack.initialPage: loginPage
 
 	// load all pages
 	Component {id: chatPage; ChatPage {}}
 	Component {id: loginPage; LoginPage {}}
 	Component {id: rosterPage; RosterPage {}}
 
-	// diconnect from jabber server
-	onClosing: kaidan.mainDisconnect()
+	// when the window was closed, disconnect from jabber server
+	onClosing: {
+		kaidan.mainDisconnect()
+	}
+
+	Component.onCompleted: {
+		function openLoginPage() {
+			// disconnect this func; the login page will do that
+			kaidan.connectionStateDisconnected.disconnect(openLoginPage);
+			// open login page
+			pageStack.pop(); // pop all pages
+			pageStack.push(loginPage);
```

Comment:
@JBBgameich 
Yep, it's fixed. Only the latest page was replaced by the login page.. Now it first pops all pages and then pushes the login page.